docs/ratt-transform/index.md

```diff
@@ -675,84 +675,384 @@ app.use(
-You can see the structure of the record inside  ```forEach()``` using ```logRecord()```:
+This results in the following assertions:
 
-```sh
-mw.forEach(
-'properties',
-mw.logRecord()
+```trig
+country:0 rdfs:label 'The Netherlands'@en.
+country:1 rdfs:label 'Germany'@en.
+country:2 rdfs:label 'Italy'@en.
+```
+
+<h4 id='parent-key'>Parent key (`$parent`)</h4>
+
+When `mw.forEach` iterates through a list of elements, it makes the enclosing *parent* record available under key `$parent`.
+
+The parent record is the record that directly contains the path that was specified in the first argument to the `mw.forEach` call.
+
+For example, the parent record in the following call is the record that directly contains the `"data"` key:
+
+```ts
+app.use(
+  mw.forEach('data.countries',
+    …
 )
 ```
-The printed result is:
 
+The `$parent` key can be observed when `mw.logRecord` is used to print the iterated-over elements to the terminal:
+
+```ts
+app.use(
+  mw.forEach('data.countries',
+    mw.logRecord()),
+)
 ```
+
+For our example source data, this emits the following 2 RATT records:
+
+```json
 {
-  "type": "Apartment",
-  "country": "Netherlands",
+  "id": "en",
+  "name": "The Netherlands",
   "$index": 0,
   "$parent": {
-    "name": "J.D.",
-    "properties": [
+    "data": {
+      "labels": [
+        {
+          "id": "en",
+          "name": "The Netherlands",
+        },
+        {
+          "id": "de"
+          "name": "Germany",
+        }
+      ]
+    }
+  },
+  "$root": "__circular__"
+}
+```
+
+and:
+
+```json
+{
+  "id": "de",
+  "name": "Germany",
+  "$index": 1,
+  "$parent": {
+    "data": {
+      "labels": [
+        {
+          "id": "en",
+          "name": "The Netherlands",
+        },
+        {
+          "id": "de"
+          "name": "Germany",
+        }
+      ]
+    }
+  },
+  "$root": "__circular__"
+}
+```
+
+The `$root` key is explained in [the next section](#root-key).
+
+<h4 id='root-key'>Root key (`$root`)</h4>
+
+Sometimes it may be necessary to access a part of the original RATT record that is outside of the scope of the `mw.forEach` call.
+
+Every RATT record inside an` mw.forEach` call contains the `"$root"` key.  The value of the root key provides a link to the full RATT record.  Because the `$root` key is part of the linked-to RATT record, it is not possible to print the value of the root key.  (This would result in infinite output.)  For this reason, the value of the `$root` key is printed as the special value `"__circular__"`.
+
+For the above examples, the parent record and root record are the same, but this is not always the case.  Specifically, the parent record and root record are different when `mw.forEach` calls are nested.
```

Comment:
The example of nested forEach is complicated, I find it hard to follow. I would propose to use different data. If we use the above data : 
```
{
  "metadata": {
    "title": {
      "name": "Data about countries."
    }
  },
  "data": {
    "countries": [
      {
        "id": "nl",
        "name": "The Netherlands"
      },
      {
        "id": "de",
        "name": "Germany"
      }
    ]
  }
}
```
and iterate over 'data.countries', the `$parent `would be data but the `$root` will show the whole record including the sibling 'metadata'. Right?

Reply:
```

Reply:
Unfortunately, the more complex example is necessary in this case.  This is left as-is.

Reply:
you're right! My mistake!